### PR TITLE
stash: optimize postgres stash a bit

### PR DESCRIPTION
**Pull request by mjibson**
Benchmarked against a local cockroach. Hopefully results are better in production, because locally I didn't see much improvement.

### Motivation
  * This PR fixes a recognized bug. https://github.com/MaterializeInc/materialize/issues/14018

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

**Comment by mjibson** [comment]
Removed a commit that changed the schema to use a cockroach-only function. That can come back after we've fully moved over to cockroach in testdrive.

File: src/stash/src/postgres.rs [hunk 1]
```diff
@@ -65,6 +66,21 @@ CREATE TABLE uppers (
...
+    compact: Statement,
+    consolidate_consolidation: Statement,
+    consolidate_insert: Statement,
+    consolidate_delete: Statement,
+    update_many: Statement,
+}
+
 /// A Stash whose data is stored in a Postgres database. The format of the
 /// tables are not specified and should not be relied upon. The only promise is
 /// stability. Any changes to the table schemas will be accompanied by a clear
```
Comment (jkosh44) [inline src/stash/src/postgres.rs:86]: I guess this was already there, but it looks like this comment was cut off?

**Review by jkosh44 — APPROVED** [review]
LGTM

File: src/stash/src/postgres.rs (hunk 1 above)
Comment (mjibson) [reply src/stash/src/postgres.rs:86]: Will fix in another PR.

**Pull request merged**